lib/SymbolGraphGen/SymbolGraph.cpp

```diff
@@ -64,10 +65,35 @@ PrintOptions SymbolGraph::getDeclarationFragmentsPrintOptions() const {
 
   Opts.ExclusiveAttrList.clear();
 
-#define DECL_ATTR(SPELLING, CLASS, OPTIONS, CODE) Opts.ExcludeAttrList.push_back(DAK_##CLASS);
-#define TYPE_ATTR(X) Opts.ExcludeAttrList.push_back(TAK_##X);
+  llvm::StringMap<AnyAttrKind> ExcludeAttrs;
+
+#define DECL_ATTR(SPELLING, CLASS, OPTIONS, CODE) \
+  if (StringRef(#SPELLING).startswith("_")) \
+    ExcludeAttrs.insert(std::make_pair("DAK_" #CLASS, DAK_##CLASS));
+#define TYPE_ATTR(X) ExcludeAttrs.insert(std::make_pair("TAK_" #X, TAK_##X));
 #include "swift/AST/Attr.def"
 
+  // Allow the following type attributes:
+  ExcludeAttrs.erase("TAK_autoclosure");
+  ExcludeAttrs.erase("TAK_convention");
+  ExcludeAttrs.erase("TAK_noescape");
+  ExcludeAttrs.erase("TAK_escaping");
+  ExcludeAttrs.erase("TAK_inout");
```

Comment:
Where are these inserted in `ExcludeAttrs`?

Also, what do we do with other attributes, like `@objc`, `@main`, `@dynamicMemberLookup`, etc.?

Reply:
These are not shown anywhere in the docs that I can see. The rest do occur from time to time, so they are not excluded here.

Reply:
`@propertyWrapper` is shown, for example: https://developer.apple.com/documentation/swiftui/state

I think others attributes might be shown, too.

Reply:
`@propertyWrapper` is a declaration attribute, not a type attribute. Declaration attributes are allowed by default here.

Reply:
(Except for underscored attributes of course)